packages/docs/developer-resources/contractkit/odis.md

```diff
@@ -0,0 +1,113 @@
+```ts
+  const authSigner: AuthSigner = {
+    authenticationMethod: OdisUtils.Query.AuthenticationMethod.ENCRYPTION_KEY,
+    rawKey: privateDataKey,
+  }
+```
+
+Otherwise, you should use the `WalletKeySigner`:
+```ts
+  const authSigner: AuthSigner = {
+    authenticationMethod: OdisUtils.Query.AuthenticationMethod.WALLET_KEY,
+    contractKit,
+  }
+```
+
+## Service Context 
+The `ServiceContext` object provides the ODIS endpoint URL and the ODIS public key (same as above).
+```ts
+const serviceContext: ServiceContext = {
+    odisUrl,
+    odisPubKey,
+  }
+```
+
+The ODIS endpoint URL for each environment can be found here:
+| Environment  | Key  | 
+|---|---|
+|  Alfajores Staging | https://alfajoresstaging-dot-celo-testnet.wl.r.appspot.com |
+| Alfajores  | https://us-central1-celo-phone-number-privacy.cloudfunctions.net |
+| Mainnet  |  https://us-central1-celo-odis-mainnet.cloudfunctions.net |
+
+The ODIS public key for each environment can be found here:
+| Environment  | Key  | 
+|---|---|
+|  Alfajores Staging | 7FsWGsFnmVvRfMDpzz95Np76wf/1sPaK0Og9yiB+P8QbjiC8FV67NBans9hzZEkBaQMhiapzgMR6CkZIZPvgwQboAxl65JWRZecGe5V3XO4sdKeNemdAZ2TzQuWkuZoA  |
+| Alfajores  |  kPoRxWdEdZ/Nd3uQnp3FJFs54zuiS+ksqvOm9x8vY6KHPG8jrfqysvIRU0wtqYsBKA7SoAsICMBv8C/Fb2ZpDOqhSqvr/sZbZoHmQfvbqrzbtDIPvUIrHgRS0ydJCMsA |
+| Mainnet  |  FvreHfLmhBjwxHxsxeyrcOLtSonC9j7K3WrS4QapYsQH6LdaDTaNGmnlQMfFY04Bp/K4wAvqQwO9/bqPVCKf8Ze8OZo8Frmog4JY4xAiwrsqOXxug11+htjEe1pj4uMA |
+
+## Query phone number identifier
+This call consumes quota. When the user runs out of quota, it's recommended to prompt the user to "purchase" more quota by sending a transaction to themselves. This method returns the pepper retrieved from the service as well as the the computed on-chain identifier that is generated using this pepper and the phone number.
+
+### BLS Blinding Client
+It's important for user privacy that the ODIS servers don't have the ability to view the raw phone number. Before making the request, the library first blinds the phone number using a BLS library. This prevents the ODIS from being able to see the phone number but still makes the resulting signature recoverable to the original phone number. The blinding client is written in Web Assembly, which is not compatible with React native. If you choose not to pass in a `BLSBlindingClient` it will default to the Web Assembly version. You may create a `ReactBlindingClient` by calling the constructor with the ODIS public key:
+
+```ts
+const blsBlindingClient = new ReactBlsBlindingClient(odisPubKey)
+```
+Or use the `WasmBlsBlindingClient` if your runtime environment supports Web Assembly:
+```ts
+const blsBlindingClient = new WasmBlsBlindingClient(odisPubKey)
+```
+
+Now you're ready to get the phone number identifier. Call `OdisUtils.PhoneNumberIdentifier.getPhoneNumberIdentifier` and pass in the following parameters:
+| Parameter  | Type | Required | Description  |
```

Comment:
I think the auto-gen docs are better suited to this kind of info. Param lists can quickly get out of date.